### ISSUE TITLE
[BUG] - VNAV continuous descend profile not followed by AP

### ISSUE THREAD

**Issue by kdries** [invernyx/md11-bugs#817]
### Describe the bug

AP in profile mode does not follow/calculate a cantinuous descend path based on STAR restrictions.

### Steps to reproduce

- Enter STAR and approach data for the destination
- Once ToD is reached, start descend with profile mode, MFS speed and NAV
- Autopilot does not follow a continuous descend path, instead does a "dive-and-drive" to the calculated altidude of next waypoint
- Repeats at next waypoint until programed altitude or restriction is reached, then same on further descend

Example: LIMC EVRIP 3E, ILS Z 35R via INLER

### Expected behaviour

Autopilot follows a continuous descend profile calculated by vnav.

### Evidence

![image](https://github.com/invernyx/md11-bugs/assets/6001596/bb656aa1-eb9d-44fc-8755-509cea55f598)
![image](https://github.com/invernyx/md11-bugs/assets/6001596/3c15dea9-c220-4d7a-bdb1-e7bfcb8faa80)

### Simulator

Microsoft Flight Simulator 2020 (MSFS)

### Crash

No

### Phase of flight

Descent

### Departure

EBBR

### Arrival

LIMC

### Cruise level

FL350

### ZFW

202560kg

### CG

40%

### Fuel

9800kg

### Cost index

200

### Route

EBBR/25R N0478F350 SOPO3G SOPOK Y863 ROPUV L607 UTABA M738 NATAG/N0490F290 M738 LORLO Y740 ATPED M984 LUSIL W323 EVRIP EVRI3E ILSZ35R LIMC/35R

### Location

Initial descend towards EVRIP3E and descend via STAR

### Agreement

- [X] Yes

**Comment by LKNprogrammer** [comment]
I would agree with this bug report.  I had similar behavior today when flying from KLAS to KOAK.  Also, the autopilot did not calculate ToD correctly.  The plane nosedived after passing the RUSME waypoint to meet the speed and altitude restriction of the next RNAV waypoint, BIFFY.

**Comment by turbofandude** [comment]
This should be fixed by an improvement in the leg progress system in 0.30+
